lib/url.ts

```diff
@@ -442,11 +443,22 @@ export default class URL implements IURL {
           this._username = this._href.slice(start, portIndex || i);
           this._password = this._href.slice(portIndex + 1, i);
           return this._extractHostname(i + 1, end);
+        } else if (code < 48 || code > 57) {
+          // non numeric character in port
+          nonNumeric = true;
         }
       }
       if (!this._port) {
         this._port = this.href.slice(portStart, i);
       }
+      // validate port - cannot contain non-numeric characters
+      if (nonNumeric) {
+        throw new TypeError('Invalid URL: port contains non numeric character');
+      }
+      // cannot be greater than 65535
+      if (this._port.length >= 5 && parseInt(this._port, 10) > 65535) {
```

Comment:
What about using `Number(this._port)` instead? It's a bit more strict than `parseInt`.